geopandas/geodataframe.py

```diff
@@ -1673,7 +1678,33 @@ def dissolve(
 
         # Process non-spatial component
         data = self.drop(labels=self.geometry.name, axis=1)
-        aggregated_data = data.groupby(**groupby_kwargs).agg(aggfunc)
+        with warnings.catch_warnings(record=True) as record:
+            warnings.filterwarnings(
+                "always",
+                message="The default value of numeric_only",
+                category=FutureWarning,
+            )
+            aggregated_data = data.groupby(**groupby_kwargs).agg(aggfunc, **kwargs)
+        for w in record:
+            if str(w.message).startswith("The default value of numeric_only"):
+                msg = (
```

Comment:
What about sticking as close to the original text as we can? Something like this? The proposed warning here is not super clear imho.

> The default value of numeric_only in aggfunc='{aggfunc}' within pandas.DataFrameGroupBy used in dissolve is deprecated. In pandas 2.0, numeric_only will default to False. Either specify numeric_only or select only columns which should be valid for the function.